How to toggle password field in vuejs (#337)
hetzneronline/community-content: tutorials/how-to-toggle-password-field-in-vuejs/01.en.md

---
SPDX-License-Identifier: MIT
path: "/tutorials/how-to-toggle-password-field-in-vuejs"
slug: "how-to-toggle-password-field-in-vuejs"
date: "2021-06-15"
title: "How To toggle the visibility of a password in an input field in Vue.js"
short_description: "Toggle the visibility of a password field in Vue.js"
tags: ["Development", "JavaScript", "Vue.js"]
author: "Thomas Boehringer"
author_link: "https://github.com/tomboe311"
author_img: "https://avatars.githubusercontent.com/u/25898733&v=4"
author_description: ""
language: "en"
available_languages: ["en", "de"]
header_img: "header-3"
cta: "development"
---

## Introduction

[Vue.js](https://vuejs.org/) is a client-side Javascript web framework, comparable to React.js or Angular.

This tutorial shows how to create reusable "components" in Vue.js, using a password field with a visibility button.

Disclaimer: This tutorial was written and tested for Vue.js 3.0.0

**Prerequisites**

* Basic knowledge about [Vue.js](https://vuejs.org/)
* You will also need a basic knowledge of JavaScript, HTML, and CSS

## Step 1 - Setting Up the Project

> At the beginning, the directory structure of the project is creating with the `mkdir` command:

Reviewer: ```suggestion
At the beginning, the directory structure of the project is created with the `mkdir` command:
```

    </head>

    <body>
        <!-- Add content in Step 2 and 3 -->
    </body>
</html>
```

## Step 2 - Create Vue.js App

We create a file `main.js` in the directory `src` with the `touch` command:

```bash
touch main.js
```

Open the file and add the following data to create a Vue App without properties:

```javascript
const app = Vue.createApp({})
```

At the end of the body in `index.html` we load the Vue App and connect it to the previously created `div` block via the ID `app`:

```javascript
<script src="./main.js"></script>
<script>
    app.mount("#app");
</script>
```

## Step 3 - Create the Component

Components are reusable instances with their own name. The components can be used as custom elements within a root instance.

### Step 3.1 - Preparing

For a clear structure we create a directory for the components and our first component file with the name `PasswordRevealer.js`:

```bash
mkdir components
touch components/PasswordRevealer.js
```

### Step 3.2 - Create Component

This component `password-revealer` renders a password input field with a reveal button.
* Create the variable `passwordType` and set the value to `"password"`.

> * When the `<button` is clicked, execute the method `switchVisibility` and change the `type` in `<input>` between `"text"` and `"password"`.

Reviewer: ```suggestion
* When the `<button>` is clicked, the method `switchVisibility` is executed and toggles the `type` in `<input>` between `"text"` and `"password"`.
```

    data() {
        return {
            passwordType: 'password'
        };
    },
    methods: {
        switchVisibility() {
            this.passwordType = this.passwordType === 'password' ? 'text' : 'password'
        }
    },
    template:
        /*html*/
        `
        <input id="password" :type="passwordType">
        <button v-on:click="switchVisibility()">Show/Hide</button>
        `
})
```

### Step 3.3 - Use Component

At the end of the body in the `index.html` the created file from step 3.2 is included as follows:

```html
<body>
  <!-- other code block -->
  <script src="./main.js"></script>
  <script src="./components/PasswordRevealer.js"></script>
  <!-- other code block -->
</body>
```

Now the component can be used within the root instance via the name `password-revealer`:

```html
<div id="app">
  <password-revealer></password-revealer>
</div>
```

## Step 4 - Styling (Optional)

For the styling of the button we create our stylesheet `my-style.css`:

```bash
mkdir -p assets/css
touch assets/css/my-style.css
```

In this file we add the CSS class `revealer`:

```css
.revealer {
    color: #fff;
    background-color: #6c757d;
    border-color: #6c757d;
}
```

> The stylesheet is including in the `<head>` in the `index.html` file:

Reviewer: ```suggestion
The stylesheet is included in the `<head>` in the `index.html` file:
```
